### PR TITLE
feat(ui): Embed Support

### PR DESCRIPTION
**Pull request by JoltCode**
- feat(ui): embed support

# Description

Adds support for Discord Embeds

Fixes N/A (issue)

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

N/A

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works 
- [x] I have updated .env.example if I added a new environment variable
- [x] My PR title follows the [semantic commits](https://www.conventionalcommits.org/en/v1.0.0/) style

### CODE REVIEW

File: packages/ui/src/components/primitives/message/Embed.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import { MessageProps } from './props';
+import { cn } from 'packages/ui/src/utils/utils';
+import { BlueLink } from '../ui/blue-link';
+
+const EmbedText = (props: {
+	children: React.ReactNode;
+	className?: string;
+}) => (
+	<span className={cn(`font-light text-primary`, props.className)}>
+		{props.children}
```
Comment (RhysSullivan) [inline packages/ui/src/components/primitives/message/Embed.tsx:11]: How does this handle discord embeds? I think if you put in our existing embed parsing logic it should work here

File: packages/ui/src/components/primitives/message/Embed.tsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+				) : (
+					embed.author?.name
+				)}
+			</EmbedText>
+			<EmbedText className="text-xl font-bold">{embed.title}</EmbedText>
+			<EmbedText>{embed.description}</EmbedText>
+			{embed.fields?.map((data, dataIteration) => (
+				<EmbedField {...data} key={`field-${dataIteration}`} />
+			))}
+			<EmbedText className="text-sm">{embed.footer?.text}</EmbedText>
```
Comment (RhysSullivan) [inline packages/ui/src/components/primitives/message/Embed.tsx:58]: Does this support stuff like embed image?

File: packages/ui/src/components/primitives/message/Embed.tsx (hunk 1 above)
Comment (JoltCode) [reply packages/ui/src/components/primitives/message/Embed.tsx:12]: Fixed!

File: packages/ui/src/components/primitives/message/Embed.tsx (hunk 2 above)
Comment (JoltCode) [reply packages/ui/src/components/primitives/message/Embed.tsx]: Not yet, I was thinking that should be for a future PR, thoughts?

Comment (RhysSullivan) [reply packages/ui/src/components/primitives/message/Embed.tsx]: sure ill open a ticket

File: packages/ui/src/components/primitives/message/Embed.tsx [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+}) => (
+	<span className={cn(`font-light text-primary`, props.className)}>
+		{props.children}
+	</span>
+);
+
+const EmbedField = async (
+	props: NonNullable<EmbedProps['embed']['fields']>[number],
+) => {
+	const nameHTML = await parse(props.name);
```
Comment (RhysSullivan) [inline packages/ui/src/components/primitives/message/Embed.tsx:19]: Move these awaits into the embed?

File: packages/ui/src/components/primitives/message/Embed.tsx [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+const numberToHex = (number: number | undefined) => {
+	if (!number) return undefined;
+	const hex = number.toString(16);
+	return hex.length === 1 ? '0' + hex : hex;
+};
+
+export const Embed = async (props: EmbedProps) => {
+	const { embed } = props;
+	const embedAuthorNameHTML = await parse(embed.author?.name ?? '');
```
Comment (RhysSullivan) [inline packages/ui/src/components/primitives/message/Embed.tsx:42]: Seems like the parsing should happen in the component

**Review by RhysSullivan — CHANGES_REQUESTED** [review]

File: packages/ui/src/components/primitives/message/Embed.tsx (hunk 3 above)
Comment (JoltCode) [reply packages/ui/src/components/primitives/message/Embed.tsx]: Fixed!

File: packages/ui/src/components/primitives/message/Embed.tsx (hunk 4 above)
Comment (JoltCode) [reply packages/ui/src/components/primitives/message/Embed.tsx]: Fixed!

**Pull request merged**